Parsing JSON Reports to allow Ignored Warnings to be displayed

tmobaird opened the pull request:
Sorry for this taking so long, but this ended up being a fairly large PR. This PR is in reference to #8 about reading JSON reports and displaying ignored warnings (while remaining backwards compatible to read the tabs reports as well). My strategy to accomplish this was to check the report file content and check if it matches a JSONObject. If it does, it will scan the file with the JSON strategy, if not it will scan the file based on the tabs strategy. 

I also had to make sure that these new ignored warnings that can be added would not affect the status of a build (ex. a build won't fail if a threshold is set for 0 warnings, and a brakeman report includes 1 ignored warning). This would've probably made a lot of developers mad if their builds started failing because of this; so I made `evaluateStatus` only factor for non-ignored annotations/warnings.

Everything is working fine for me in two different local Jenkins instances!

Here are some more details about what this PR includes:
- Ability to use JSON reports (as well as tabs)
- Can read ignored warnings in JSON reports
- Annotations with category "Ignored" don't factor into threshold evaluations
- Ignored Warnings Total on build status page
- Ignored Warnings added to Reports screens and Trend Graphs
- Error handling for invalid JSON reports

This closes #8.

Feel free to let me know if there are any suggestions or concerns!

ConstantineXVI commented:
I'll review this [hopefully] sometime this weekend.

tmobaird commented:
@ConstantineXVI sounds good, thanks!

ConstantineXVI commented on src/main/java/hudson/plugins/brakeman/scanners/BrakemanJSONScanner.java:
```diff
@@ -0,0 +1,67 @@
+        }
+    }
+
+    private String getCategory(String filterType) {
```
Likewise on the enums here.

tmobaird replied:
👍

tmobaird commented on pom.xml:
```diff
@@ -55,6 +55,11 @@
+    <dependency>
+      <groupId>org.json</groupId>
+      <artifactId>json</artifactId>
+      <version>20090211</version>
```
That's my bad. I meant to update this before I created the PR. I believe it was just the most popular version, which is why I started with this. I'll update this to a newer version

tmobaird commented on src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java:
```diff
@@ -91,67 +91,71 @@ public Action getProjectAction(final AbstractProject<?, ?> project) {
+     * @param content
+	 * @return result
+	 */
+	protected String getFileType(String content) {
```
That sounds good to me. I'll update asap 👍

tmobaird commented:
Ok all issues with this PR **should** be resolved now. Here's a summary of the changes:
- Removed org.json dependency from pom, not needed because net.sf.json is available to you
- Changed org.json imports to net.sf.json, and refactored a little bit of the JSON parsing code to use this library's [net.sf.json] code
- Switched a couple methods to return an enum, per suggestion

Let me know if there's anything else that is needed.

tmobaird commented:
@ConstantineXVI any progress on this?

ConstantineXVI commented:
Sorry, I've been swamped lately. I'm hoping to finish this up over the holiday.

tmobaird commented:
@ConstantineXVI no worries, I was just curious. Thanks!

tmobaird commented:
@ConstantineXVI thanks! 👍
